core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,591 @@
+
+datastore_services = models.Registry.import_datastore_services()
+transaction_services = models.Registry.import_transaction_services()
+
+PLATFORM_CHOICE_ANDROID = 'android'
+PLATFORM_CHOICE_WEB = 'web'
+PLATFORM_CHOICES = [PLATFORM_CHOICE_ANDROID, PLATFORM_CHOICE_WEB]
+
+REPORT_INFO_TO_REDACT = (
+    'user_feedback_other_text_input', 'event_logs', 'logcat_logs')
+
+
+class AppFeedbackReportModel(base_models.BaseModel):
+    """Model for storing feedback reports sent from learners.
+
+    Instances of this model contain information about learner's device and Oppia
+    app settings, as well as information provided by the user in the feedback
+    report.
+
+    The id of each model instance is determined by concatenating the platform,
+    the timestamp of the reports (in ms since epoch, in UTC), and a hash of a
+    string representation of a random int.
+    """
+
+    # We use the model id as a key in the Takeout dict.
+    ID_IS_USED_AS_TAKEOUT_KEY = True
+
+    # The platform (web or Android) that the report is sent from and that the
+    # feedback corresponds to.
+    platform = datastore_services.StringProperty(
+        required=True, indexed=True, choices=PLATFORM_CHOICES)
+    # The ID of the user that scrubbed this report, if it has been scrubbed.
+    scrubbed_by = datastore_services.StringProperty(
+        required=False, indexed=True)
+    # Unique ID for the ticket this report is assigned to (see
+    # AppFeedbackReportTicketModel for how this is constructed). This defaults
+    # to None since initially, new reports received will not be assigned to a
+    # ticket.
+    ticket_id = datastore_services.StringProperty(default=None, indexed=True)
+    # Datetime in UTC of when the report was submitted by the user on their
+    # device. This may be much earlier than the model entity's creation date if
+    # the report was locally cached for a long time on an Android device.
+    submitted_on = datastore_services.DateTimeProperty(
+        required=True, indexed=True)
+    # The type of feedback for this report; this can be an arbitrary string
+    # since iterations of the report structure may introduce new types and we
+    # cannot rely on the backend updates to fully sync with the frontend report
+    # updates.
+    report_type = datastore_services.StringProperty(required=True, indexed=True)
+    # The category that this feedback is for.
+    category = datastore_services.StringProperty(required=True, indexed=True)
+    # The version of the app; on Android this is the package version name (e.g.
+    # 1.0-release-arm...) and on web this is the release version (e.g. 3.0.8).
+    platform_version = datastore_services.StringProperty(
+        required=True, indexed=True)
+    # The user's country locale represented as a ISO-3166 code; the locale is
+    # determined by the user's Android device settings.
+    device_country_locale_code = datastore_services.StringProperty(
```

Comment:
Should this be ``android_device_country_locale_code`` per Sean's comment below?

Reply:
Ah I had thought this could be determined on Web as well but it looks like it just uses language locale as opposed to country locale. Will specify android

Reply:
Done